scripts/eqtl_hail_batch/launch_conditional_analysis.py

```diff
@@ -0,0 +1,160 @@
+"""
+Launch analysis runner for all cell types and chromosomes
+
+For example:
+
+    python3 scripts/hail_batch/eqtl_hail_batch/launch_conditional_analysis.py \
+        --input-path "gs://cpg-tob-wgs-test/scrna_seq/grch38_association_files" \
+        --dry-run \
+        --output-dir gs://cpg-tob-wgs-test/eqtl_output \
+        --chromosomes '1 2'
+"""
+
+import logging
+import os
+import click
+
+from analysis_runner.cli_analysisrunner import run_analysis_runner
+from google.cloud import storage
+
+
+# def run_analysis_runner(*args, **kwargs):
+#     """Run analysis runner for all cell types and chromosomes"""
+#     print(args, kwargs)
+
+
+@click.command()
+@click.option(
+    '--cell-types',
+    default=None,
+    help='List of cell types to test',
+)
+@click.option(
+    '--chromosomes',
+    required=True,
+    help='List of chromosome numbers to run eQTL analysis on. Space separated, as one argument',  # noqa: E501; pylint: disable=line-too-long
+)
+@click.option(
+    '--input-path',
+    required=True,
+    help=(
+        'A path prefix of where input files are located, eg: gs://MyBucket/folder/. '
+    ),
+)
+@click.option(
+    '--first-round-path',
+    required=True,
+    help=(
+        'A path prefix of where the output from the first round of esnp '
+        'results are located, eg: gs://MyBucket/folder/. '
+    ),
+)
+@click.option(
+    '--output-dir',
+    required=True,
+    help='A path of where to output files, eg: gs://MyBucket/output-folder/',
+)
+@click.option(
+    '--test-subset-genes',  # pylint: disable=too-many-locals
+    type=int,
+    help='Test with {test_subset_genes} genes, often = 5.',
+)
+@click.option('--dry-run', is_flag=True, help='Just check if files exist')
+def submit_eqtl_jobs(
+    chromosomes,
+    input_path,
+    first_round_path,
+    output_dir,
+    test_subset_genes=None,
+    dry_run=False,
+    cell_types=None,
+):
+    """Run association script for all chromosomes and cell types"""
+
+    assert output_dir.startswith('gs://') and input_path.startswith('gs://')
+    chromosomes = chromosomes.split(' ')
+    assert isinstance(chromosomes, list)
+
+    cs_client = storage.Client()
+
+    bucket_name = input_path.split('gs://')[1].split('/')[0]
+    bucket = cs_client.get_bucket(bucket_name)
+    bucket_path = input_path.split(f'gs://{bucket_name}/')[-1]
+
+    def file_exists(files):
+        file_status = storage.Blob(
+            bucket=bucket, name=files.split(bucket_name)[-1].strip('/')
+        ).exists(cs_client)
+        return file_status
```

Comment:
This is a good use case for cloudpathlib!

you can do a cheeky 
`from cloudpathlib import AnyPath`

then 
`if AnyPath(file).exists(): ... `

Reply:
https://cloudpathlib.drivendata.org/stable/#quick-usage
This can also be used to list contents of a local or remote directory!

```
p = list(AnyPath('gs://cpg-acute-care-test/reanalysis').glob('*'))
p[0].exists()
```

Reply:
Awesome, I didn't know you can do this in cloudpathlib. Does one have an advantage over another, or are these just different options?

Reply:
Just that exists() can be used as a replacement for the file_exists method, and the glob can be used instead of `bucket.list_blobs(prefix=path_to_expression_files + '/', delimiter='/')`!

Reply:
I just came back to this because I needed it as a code example, and realised what a pointless response this was ^^

Advantages with cloudpathlib are that `CloudPaths` or `AnyPaths` are cloud-provider agnostic. The library is built to recognise what a GCP/AWS/Azure path looks like, and will handle everything appropriately behind the scenes. `AnyPath` also allows a mix of local and cloud paths, and it will work out how to access the file.

On one hand it's cleaner code, because you don't have to open a bucket/client connection etc. etc., but it's also potentially better code, as we can switch to any other cloud provider and no code changes are required.